pandas/core/groupby/generic.py

```diff
@@ -584,13 +584,14 @@ def _choose_path(self, fast_path, slow_path, group):
             res_fast = fast_path(group)
 
             # compare that we get the same results
+            if res_fast.columns != group.columns:
+                return path, res
             if res.shape == res_fast.shape:
```

Comment:
can you add a comment here on what this is checking

Reply:
Done.